src/TableUtils.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * Utility methods to perform read/write operations on a table element.
+ */
+var TableUtils = {
+
+  createTableCell: function() {
+    var tableCell = document.createElement('td');
+    return tableCell;
+  },
+
+  createTableRow: function(columnCount) {
+    var tableRow = document.createElement('tr');
+    for (var i = 0; i < columnCount; i++) {
+      tableRow.appendChild(TableUtils.createTableCell());
+    }
+
+    return tableRow;
+  },
+
+  createTable: function(rowCount, columnCount) {
+    var table = document.createElement('table');
+    var tableBody = document.createElement('tbody');
+
+    for (var i = 0; i < rowCount; i++) {
+      tableBody.appendChild(TableUtils.createTableRow(columnCount));
+    }
+
+    table.appendChild(tableBody);
+    return table;
+  },
+
+  getColumnCount: function(table) {
+    var body = table.childNodes[0];
+    var rows = body.childNodes;
+    return rows[0].childNodes.length;
+  },
+
+  getRowCount: function(table) {
+    var body = table.childNodes[0];
+    return body.childNodes.length;
+  },
+
+  // Finder operations
+
+  findTable: function(scribe, el) {
+    while (el && el !== scribe.el && el.nodeName !== 'TABLE') {
+      el = el.parentNode;
+    }
+
+    if (el && el.nodeName !== 'TABLE') {
+      el = null;
+    }
+
+    return el;
+  },
+
+  findTableCell: function(scribe, el) {
+    while (el && el !== scribe.el && el.nodeName !== 'TD') {
+      el = el.parentNode;
+    }
+
+    if (el && el.nodeName !== 'TD') {
+      el = null;
+    }
+
+    return el;
+  },
+
+  findCellPosition: function(table, targetTableCell) {
+    var body = table.childNodes[0];
+    var rows = Array.prototype.slice.call(body.childNodes);
+    var position = {
+      rowIndex: -1,
+      columnIndex: -1
+    };
+
+    rows.every(function(row, rowIndex) {
```

Comment:
oh i see, you want to exit early from the loop the first time this returns false. got it.